### PR TITLE
Issue 495: Add default values for query builder inside connection settings

### LINKED ISSUE #495: Ability to set a default column names for the primary time column in the datasource config. for Date and Timestamp, This would be used by the $timeFilter etc. macros unless overridden in the query builder
add to ConfigEditor
- datetime
- datetime64
- uint32
- date / date32

### PR DESCRIPTION
**Pull request by lunaticusgreen**
fix #495

### CODE REVIEW

**Comment by antip00** [comment]
Issues:

- [ ] Can't create new datasource, `save and test` button returns nothing
Screenshot from 2024-05-31 16-20-53
Screenshot from 2024-05-31 16-21-09



- [ ] I think we need to move `default database` to other section
Screenshot from 2024-05-31 16-21-09

File: src/views/ConfigEditor/FormParts/DefaultValues/DefaultValues.tsx [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+    >
+      <InlineSwitch
+        id="useDefaultConfiguration"
+        className="gf-form"
+        value={jsonData.useDefaultConfiguration || false}
+        onChange={(e) => onSwitchToggle('useDefaultConfiguration', e.currentTarget.checked)}
+      />
+    </InlineField>
+    {jsonData.useDefaultConfiguration && <>
+      {/*<h6>Connection Parameters</h6>*/}
```
Comment (Slach) [inline src/views/ConfigEditor/FormParts/DefaultValues/DefaultValues.tsx:86]: remove unused code

File: src/views/QueryEditor/QueryEditor.tsx [hunk 2]
```diff
@@ -125,22 +135,47 @@ export function QueryEditor(props: QueryEditorProps<CHDataSource, CHQuery, CHDat
...
     intervalFactor: query.intervalFactor || DEFAULT_INTERVAL_FACTOR,
     interval: query.interval || '',
     query: query.query || defaultQuery,
     formattedQuery: query.formattedQuery || query.query,
-    editorMode: EditorMode.Builder,
+    editorMode: EditorMode.Builder
   };
 
+  if (datasource.defaultValues && !query.initialized) {
+    // !!!!!!!!
```
Comment (Slach) [inline src/views/QueryEditor/QueryEditor.tsx:155]: what means this comment?

File: src/views/QueryEditor/components/QueryTextEditor/editor/initiateEditor.ts [hunk 3]
```diff
@@ -29,7 +29,9 @@ export const THEME_NAME = "clickhouse-dark-theme";
 export const LANGUAGE_ID = "clickhouseLanguage";
 
 export const initiateEditor = (templateVariables: any, monacoInstance: any, autocompletionData: any, systemDatabasesData: any) => {
-	const {Method, Variable, Constant, Keyword, TypeParameter, Text} = monacoInstance.languages.CompletionItemKind
+	const {
+Method, Variable, Constant, Keyword, TypeParameter, Text
```
Comment (Slach) [inline src/views/QueryEditor/components/QueryTextEditor/editor/initiateEditor.ts:33]: why new line without ident?

**Review by Slach — CHANGES_REQUESTED** [review]

**Comment by Slach** [comment]
could run `git pull origin master`?

**Comment by Slach** [comment]
@lunaticusgreen 
image

https://github.com/Altinity/clickhouse-grafana/actions/runs/9567400079/job/26375045064?pr=567
test failed 
"Save and Test" button return nothing, expect error, look to screenshot

**Review by Slach — APPROVED** [review]

**Pull request merged**